feat(MediaPlayer): improve handling of calls and MediaPlayer

sdwoodbury opened the pull request:
<!--  Thanks for sending a pull request!-->

### What this PR does 📖

- the controls for the current call (mute, silence, ect) were moved from State::UI to State::UI::Call
- for a video call, UI::Call stores the WebView (currently this only happens when the video is popped out). the popout_player persists when one switches chats. 
- ToggleMedia and EndAll were changed because
    1) ToggleMedia allowed for setting multiple chats to having active media but only one media should be active at a time. 
    2) EndAll only makes sense if every chat can have active media (see previous point about this) 

### Which issue(s) this PR fixes 🔨

- Resolve Satellite-im/Uplink#44 

<!--Add the ticket Github number such as #Resolve Satellite-im/Uplink#1 to automatically link the PR to the issue-->

### Special notes for reviewers 🗒️

### Additional comments 🎤
for some reason if overlays are toggled some number of times, ending with them being enabled, and the app is closed, the app crashes. But the WebViews close themselves when the UI struct is dropped. I suspect the test code in main.rs has something to do with this, and have left the problem for now in hopes that when overlays are actually used, this bug will go away.

hyj1204 reviewed (approved):
Looks great!
![image](https://user-images.githubusercontent.com/14248245/179837937-1666901b-69aa-436d-9f7c-2561b014701f.png)

sdwoodbury committed:
automatically close all windows when State is dropped

sdwoodbury committed:
create Call struct with State::UI
move active_media from Chat to Chats and make it an Option<Uuid>

sdwoodbury committed:
Merge branch 'dev' into feat/44

sdwoodbury committed:
fix(compose)

sdwoodbury committed:
fix(MediaPlayer)

sdwoodbury committed:
feat(state): allow setting and clearing the active media

sdwoodbury committed:
fix(clippy)

sdwoodbury committed:
fix(MediaPlayer): close MediaPlayer

sdwoodbury committed:
properly drop overlays
store WebView in State::UI.call

sdwoodbury committed:
track the WebView for the popout player

sdwoodbury committed:
fix(clippy)

sdwoodbury committed:
fix(popout player): don't close it when switching chats
also dont' persist Chats.active_media

sdwoodbury committed:
comment

phillsatellite commented:
I wanted to mention a little thing I found, it seems like if you click the close button multiple times, it will just force the pop out into fullscreen and wont actually exit pop out player. I showed @sdwoodbury and going to create ticket for it 🔨 

https://user-images.githubusercontent.com/93608357/210622939-1d786c57-a32f-4d76-89eb-d269c091c267.mov

InfamousVague commented:
Two things I found:

- When the call is ended, the video continues to play.
- When the video is popped out, it should display the "media detatched" component like it used to.

sdwoodbury commented:
@InfamousVague 

>     * When the call is ended, the video continues to play.
  that issue is for ticket Satellite-im/Uplink-Backlog#2. and about that - it seems to be a problem with the video tag. i'm not sure it makes sense to spend time debugging that because as far as I know, we won't be using the video tag for video streaming. Rather we will be using some new thing that the Dioxus team is working on. 

>     * When the video is popped out, it should display the "media detatched" component like it used to.
i'll fix this.

sdwoodbury committed:
fix(popout): don't display the same video twice.
strangely enough, getting rid of the video element makes the popout window not work

sdwoodbury committed:
comment

sdwoodbury commented:
using `write()` instead of `write_silent()` to create the popout causes the video to not display in the popout window. makes no sense. if `write_silent()` is used and the user then clicks on one of the chats, the UI updates, the "media detached" message is displayed in the chat, and the video continues to work in the popout.

sdwoodbury commented:
Will revisit this later
